lib/omniauth/strategies/saml.rb

```diff
@@ -103,15 +109,23 @@ def other_phase
       uid { @name_id }
 
       info do
-        {
-          :name  => @attributes[:name],
-          :email => @attributes[:email] || @attributes[:mail],
-          :first_name => @attributes[:first_name] || @attributes[:firstname] || @attributes[:firstName],
-          :last_name => @attributes[:last_name] || @attributes[:lastname] || @attributes[:lastName]
-        }
+        found_attributes = options.attribute_statements.map do |key, values|
+          attribute = find_attribute_by(values)
+          [key, attribute]
```

Comment:
Yeah, I wonder if it makes sense to address that in a separate PR.  It's starting to sound like it would be worth considering something more robust than a simple Hash for handling the various mapping scenarios...